lib/manager/nuget/extract.ts

```diff
@@ -1,10 +1,50 @@
+import * as findUp from 'find-up';
+import * as path from 'path';
+import { XmlDocument } from 'xmldoc';
+import { readFileSync } from 'fs';
 import { logger } from '../../logger';
 import { get } from '../../versioning';
 import { PackageDependency, ExtractConfig, PackageFile } from '../common';
 import * as semverVersioning from '../../versioning/semver';
 import * as datasourceNuget from '../../datasource/nuget';
 import { SkipReason } from '../../types';
 
+function determineRegistryUrls(
+  packageFile: string,
+  localDir: string
+): string[] {
+  const registryUrls = [datasourceNuget.getDefaultFeed()];
```

Comment:
Why is it necessary to return defaults? Can't we return *no* registryUrls if they're not found, and let the datasource add them?

Reply:
We need the default here anyway because when there's no `<clear>` entry in the config file, it only adds feeds to the default feed. 

At least for NuGet I would say there is no default at the datasource side , everything is controlled by client-side config.

Reply:
But there can be a `clear` or `remove`. I would suggest to set it to undefined here, if no config file exists and set the default values in datasource as @rarkins  suggests

Reply:
So if there is no `NuGet.config` or there are no `packageSources` in config, simply return  `undefined`. So we don't break existing projects.

I've configured the `registryUrls` in `config.js`, so this has to work after this pr, otherwise this would be a breaking.

Reply:
OK, I changed it so that registry URLs are only set here when there are package sources  defined in the nuget config (5dd788c881168b15294db1569eea037b4753ea30). WDYT?